tests/test_make_idl_file.py

```diff
@@ -0,0 +1,44 @@
+# This file is part of ts_idl.
+#
+# Developed for the LSST Telescope and Site Systems.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+import os
+import unittest
+
+from lsst.ts import idl
+from lsst.ts.sal.make_idl_file import make_idl_file
+
+
+class TestMakeIdlFile(unittest.TestCase):
+    def test_make_idl_file(self):
+
+        component = "Test"
+        idl_file_name = f"sal_revCoded_{component}.idl"
+        idl_file_path = idl.get_idl_dir() / idl_file_name
+
+        if os.path.exists(idl_file_path):
+            os.remove(idl_file_path)
+
+        make_idl_file(
+            name=component,
+            keep_all=False,
+        )
+
+        assert os.path.exists(idl_file_path)
```

Comment:
It makes me nervous to have this test permanently modify data in ts_idl. What if the developer is intentionally using an older version?

Pleases consider move the old file aside, if it exists, then moving it back in a `finally` clause.

The trick, I suppose, is deciding how to move the file. Perhaps using `tempfile` library would be a reasonable solution for that -- e.g. by creating a temporary directory, moving the old file there, then moving it back. The directory would be automatically cleaned up as long as you used tempfile.TemporaryDirectory as a context manager.